src/is_master.rs

```diff
@@ -31,21 +31,35 @@ pub(crate) fn is_master_command(api: Option<&ServerApi>) -> Command {
 /// Run the given isMaster command.
 ///
 /// If the given command is not an isMaster, this function will return an error.
-pub(crate) async fn run_is_master(
-    command: Command,
-    conn: &mut Connection,
-) -> Result<IsMasterReply> {
+pub(crate) async fn run_is_master(command: Command, conn: &mut Connection) -> IsMasterResult {
     if !command.name.eq_ignore_ascii_case("ismaster") && !command.name.eq_ignore_ascii_case("hello")
```

Comment:
edit: also see my other comment about possibly ditching this function (at least in the monitor) altogether.

I think this check of the command name is proving to be a bit over-paranoid and we'd probably be better off just removing it. This function is only called in two places, both of which will always use a hello-type command, and having this check here makes us have to deal with the possibility of a `None` duration. If we do want to enforce this, it'd probably be better to do so as a compile time check via a newtype rather than runtime as well, though I'm not really sure that's necessary either.